cmd/acb/commands/download/download.go

```diff
@@ -40,24 +41,41 @@ var Command = cli.Command{
 			Usage: "maximum execution time in seconds",
 			Value: 60,
 		},
+		cli.StringSliceFlag{
+			Name:  "credential",
+			Usage: "login credentials for custom registry",
+		},
 	},
 	Action: func(context *cli.Context) error {
 		var (
 			downloadCtx = context.Args().First()
 			dryRun      = context.Bool("dry-run")
 			destination = context.String("destination")
 			timeout     = time.Duration(context.Int64("timeout")) * time.Second
+			creds       = context.StringSlice("credential")
 		)
 
 		if downloadCtx == "" {
 			return errors.New("download requires context to be provided, see download --help")
 		}
 
+		// Add all creds provided by the user in the --credential flag
+		credentials, err := graph.CreateRegistryCredentialFromList(creds)
+		if err != nil {
+			return err
+		}
+
 		ctx, cancel := gocontext.WithTimeout(gocontext.Background(), timeout)
 		defer cancel()
 
 		pm := procmanager.NewProcManager(dryRun)
-		scanner, err := scan.NewScanner(pm, downloadCtx, "", destination, nil, nil, "")
+
+		registryLoginCredentials, err := graph.ResolveCustomRegistryCredentials(ctx, credentials)
```

Comment:
>ResolveCustomRegistryCredentials [](start = 41, length = 32)

Is there any perf penalty here if the source is not registry artifact? ResolveCustomRegistryCredentials may call msi and kv to get the credential? @shahzzzam

Reply:
I assume this is regarding the case where there are credentials passed in but we don't actually need those credentials because the context to download from is git or tar. In that case what I could add is to check here if the `downloadCtx` is actually a registry artifact and if it is then we can do all the credential processing.

Reply:
@northtyphoon yes, you;'re right. It will call Token server. @akashsinghal , correct. You can check for OCI, and only then pass in credentials to Scanner.